### PR TITLE
Add GitHub workflows

### PR DESCRIPTION
**Pull request by Ayodhya94**
## Purpose
Add GitHub workflows to build the project daily, at a PR, and at a PR merge to master branch

### CODE REVIEW

File: .github/workflows/build-master.yml
Comment (nilushancosta) [inline .github/workflows/build-master.yml:28]: No need to export environment variables as it is handled by the `env:` block above
```suggestion
        run: ./gradlew build
```

File: .github/workflows/pull-request.yml
Comment (nilushancosta) [inline .github/workflows/pull-request.yml:25]: No need to export environment variables as it is handled by the `env:` block above
```suggestion
        run: ./gradlew build
```

File: .github/workflows/daily-build.yml
Comment (nilushancosta) [inline .github/workflows/daily-build.yml:28]: No need to export environment variables as it is handled by the `env:` block above
```suggestion
        run: ./gradlew build
```

**Review by nilushancosta — CHANGES_REQUESTED** [review]

**Comment by nilushancosta** [comment]
`.github/workflows/pull-request.yml` triggers a build for pushes to any branch in addition to pull requests to any branch. `.github/workflows/build-master.yml` triggers a build for pushes to master branch. So builds triggered by pushes to master branch are already captured by `.github/workflows/pull-request.yml`
Is there a reason to have a separate build for pushes to master branch?

File: .github/workflows/daily-build.yml
Comment (Ayodhya94) [reply .github/workflows/daily-build.yml:5]: Should we build only once? Since Ballerina repo is building twice I thought it's the normal practice.

Comment (nilushancosta) [reply .github/workflows/daily-build.yml:5]: Since the filename and description said "dailly build" I assumed that we want to run the build once a day. Not an issue to run it more than once

**Comment by Ayodhya94** [comment]
> `.github/workflows/pull-request.yml` triggers a build for pushes to any branch in addition to pull requests to any branch. `.github/workflows/build-master.yml` triggers a build for pushes to master branch. So builds triggered by pushes to master branch are already captured by `.github/workflows/pull-request.yml`
> Is there a reason to have a separate build for pushes to master branch?

This was a mistake. Removed the keyword caused to trigger build for pushes to any branch.

File: .github/workflows/build-master.yml
Comment (miyurud) [inline .github/workflows/build-master.yml:22]: We will also have to add the following [line](https://github.com/ballerina-platform/module-ballerina-docker/blob/master/.github/workflows/daily-build.yml#L22).

```
GITHUB_TOKEN: ${{ secrets.BALLERINA_STDLIB_TOKEN }}
```
That is required to pull the stdlib artifacts from github.

**Review by miyurud — CHANGES_REQUESTED** [review]
Please see the comment

File: .github/workflows/daily-build.yml
Comment (miyurud) [reply .github/workflows/daily-build.yml:5]: I agree with @Ayodhya94 that we should follow the same approach followed by other Ballerina repos. I checked with [docker](https://github.com/ballerina-platform/module-ballerina-docker/blob/master/.github/workflows/daily-build.yml#L5), [http](https://github.com/ballerina-platform/module-ballerina-http/blob/master/.github/workflows/daily-build.yml#L5), and [IO](https://github.com/ballerina-platform/module-ballerina-io/blob/master/.github/workflows/daily-build.yml#L5) modules as well. All of them use the same cron expression. Hence lets use `0 */12 * * *` as the cron expression here.

Comment (Ayodhya94) [reply .github/workflows/daily-build.yml:5]: Okay. I'll keep this as it is.

File: .github/workflows/build-master.yml
Comment (Ayodhya94) [reply .github/workflows/build-master.yml:22]: Added this line

**Review by nilushancosta — APPROVED** [review]

File: .github/workflows/build-master.yml
Comment (miyurud) [reply .github/workflows/build-master.yml:22]: Ok

File: .github/workflows/build-master.yml
Comment (niveathika) [inline .github/workflows/build-master.yml:25]: BALLERINA_STDLIB_TOKEN is now depreciated, check latest stdlibs workflows for updated token

File: .github/workflows/pull-request.yml
Comment (niveathika) [inline .github/workflows/pull-request.yml:3]: Let's add push also since tokens are not shared with fork repo, it is best practise to send a PR form the upstream branch to master. Push event also needs to be added to see the build in the upstream branch before the merge.

**Comment by niveathika** [comment]
Also let's add the build badges to the READMe to easily identify the state of the repo

File: .github/workflows/build-master.yml
Comment (Ayodhya94) [reply .github/workflows/build-master.yml]: Updated to match the latest tokens

File: .github/workflows/pull-request.yml
Comment (Ayodhya94) [reply .github/workflows/pull-request.yml]: Added push event

**Comment by Ayodhya94** [comment]
> Also let's add the build badges to the README to easily identify the state of the repo.
> See https://github.com/ballerina-platform/module-ballerina-io/blob/master/README.md

Added the badges. We have to wait till the merge to add URLs of GitHub actions.

**Review by niveathika — APPROVED** [review]

**Review by miyurud — APPROVED** [review]

**Pull request merged**
